Feature/1183/231 azure e2e test

algattik opened the pull request:
## What this PR changes/adds

Fixes end-to-end azure storage copy process, and adds a system test to verify it.

The EDC provides the DPF with a storage access signature (rather than storage account key) for the sink. The storage access signature is passed via key vault.

## Why it does that

## Further notes

## Linked Issue(s)

Relates to #231 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

cpeeyush commented on system-tests/azure-tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/BlobTransferIntegrationTest.java:
```diff
@@ -0,0 +1,239 @@
+    private String getProvisionedContainerName() {
+        return given()
+                .baseUri(CONSUMER_CONNECTOR_MANAGEMENT_URL + CONSUMER_MANAGEMENT_PATH)
+                .log().all()
```
nit: logging
```suggestion
```

cpeeyush commented on system-tests/azure-tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/BlobTransferIntegrationTest.java:
```diff
@@ -0,0 +1,239 @@
+                    "web.http.data.port", String.valueOf(CONSUMER_MANAGEMENT_PORT),
+                    "web.http.data.path", CONSUMER_MANAGEMENT_PATH,
+                    "web.http.ids.port", String.valueOf(CONSUMER_IDS_API_PORT),
+                    "web.http.ids.path", "/api/v1/ids",
```
nit: `/api/v1/ids` can also be added constant and used like others e.g. CONSUMER_MANAGEMENT_PATH

Izzzu commented on extensions/azure/data-plane/storage/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azurestorage/pipeline/AzureStorageDataSinkFactory.java:
```diff
@@ -31,7 +34,6 @@
 import static java.lang.String.format;
 import static org.eclipse.dataspaceconnector.azure.blob.core.validator.AzureStorageValidator.validateAccountName;
 import static org.eclipse.dataspaceconnector.azure.blob.core.validator.AzureStorageValidator.validateContainerName;
-import static org.eclipse.dataspaceconnector.azure.blob.core.validator.AzureStorageValidator.validateSharedKey;
```
Can we remove method `validateSharedKey` from AzureStorageValidator? I see now it's used only in tests

Izzzu commented:
@algattik It looks like Readme in data-plane azure extension should be also updated, because it mentions in the example usage sharedKey in the properties:
https://github.com/agera-edc/DataSpaceConnector/tree/main/extensions/azure/data-plane/storage

algattik commented:
> @algattik It looks like Readme in data-plane azure extension should be also updated, because it mentions in the example usage sharedKey in the properties:
> https://github.com/agera-edc/DataSpaceConnector/tree/main/extensions/azure/data-plane/storage

Good point, I've updated the README and removed the example.

cpeeyush commented on system-tests/azure-tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/BlobTransferIntegrationTest.java:
```diff
@@ -0,0 +1,238 @@
+                .then()
+                .statusCode(200)
+                .extract().body()
+                .jsonPath().getString("[0].dataDestination.container");
```
otherwise, test fails for me.

```suggestion
                .jsonPath().getString("[0].dataDestination.properties.container");
```

Json response of /transferprocess e.g.

```
	"dataDestination": {
		"properties": {
			"container": "a6ed5b83-20d6-4869-8dd1-1a6f34218eee",
```

cpeeyush commented on system-tests/runtimes/azure-storage-transfer-consumer/build.gradle.kts:
```diff
@@ -0,0 +1,58 @@
+}
+
+tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
+    exclude("**/pom.properties", "**/pom.xm")
```
can be removed. Or if we keep then there is a typo **/pom.xml instead of .xm
```suggestion
```

cpeeyush commented on system-tests/runtimes/azure-storage-transfer-provider/build.gradle.kts:
```diff
@@ -0,0 +1,67 @@
+}
+
+tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
+    exclude("**/pom.properties", "**/pom.xm")
```
can be removed. Or if we keep then there is a typo **/pom.xml instead of .xm
```suggestion
```

cpeeyush reviewed (approved):
Approved with suggestions.

algattik commented:
Test was not running in CI. Adding it

cpeeyush commented:
Cosmos tests are failing do we already have a bug/story in backlog for that ?
//cc @Izzzu
